chapel-lang/chapel
# Rename BoundedRangeType to boundKind, range.boundedType to range.bounds

Resolves #17126

This PR implements the following renamings, as per https://github.com/chapel-lang/chapel/issues/17126#issuecomment-1421875075 and deprecates the old names:

| old | new |
| --- | --- |
| range.`boundedType` | range.`bounds` |
| enum `BoundedRangeType` | enum `boundKind` |
| enum constants: | |
| `bounded` | `both` |
| `boundedLow` | `low` |
| `boundedHigh` | `high` |
| `boundedNone` | `neither` |

Key changes:
* support for the redirections in ChapelRange.chpl and the compiler
* switch to the new names in:
  - modules
  - tests
  - compiler
  - spec
* add a deprecation test

While there:
* add future tests for #22042
* remove `test/io/formatters/basic-types.good` because it is unused and is identical to .default.good
* simplify copying of maps in a UseStmt constructor
* define the terms "bounded range" and "unbounded range" in the spec
* other minor tweaks

<details><summary>click to see the script I used to do the bulk of the renamings</summary>

```python3
import sys, re;

filename = sys.argv[1]

try:
  with open(filename, 'r') as fp:
    lines=fp.read()
except Exception as e:
  print(filename)
  print(e)
  sys.exit(1)

orig = lines

def bb0(m):
  print(m.group(), "||", m.groups())
  return "xxx"

def bb(m):
  if m.group() == "BoundedRangeType": return "boundKind"
  if m.group(2) == "":     return "boundKind.both"
  if m.group(2) == "Low":  return "boundKind.low"
  if m.group(2) == "High": return "boundKind.high"
  if m.group(2) == "None": return "boundKind.neither"
  print(f"bb: unknown <{m.group()}>")
  sys.exit(1)

def rr(m):
  if m.group(2) == "": return f"{m.group(1)},both,"
  if m.group(2) == "Low": return f"{m.group(1)},low,"
  if m.group(2) == "High": return f"{m.group(1)},high,"
  if m.group(2) == "None": return f"{m.group(1)},neither,"
  print(f"rr: unknown <{m.group()}>")
  sys.exit(1)

lines = re.sub(r'\bBoundedRangeType(.bounded(|Low|High|None)|)\b', bb, lines) lines = re.sub(r'\b(range\(u?int\(\d+\)),bounded(|Low|High|None),',rr, lines) lines = re.sub(r'\bboundedType\b', 'bounds', lines)

if lines != orig:
  print(filename)
  with open(filename, 'w') as fp:
    fp.write(lines)
```

</details>

Detailed dev history: e0178ad8a5..03d3bea60b and acd40ed45a..66fff36055

* [x] testing: standard and gasnet paratests; test-frontend
